### PR TITLE
disseminate downing decisions faster, #29612

### PR DESCRIPTION
**Pull request by patriknw**
* when SBR downs the reachable side (minority) it's important
  to quickly inform everybody to shutdown
* send gossip directly to downed node, STONITH signal
* gossip to a few random immediatly when self is downed, which
  is always the last from the SBR downing
* enable gossip speedup when there are downed members

Refs #29612

### CODE REVIEW

File: akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala [hunk 1]
```diff
@@ -945,6 +945,13 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef, joinConfigCompatCh
         val newGossip = localGossip.markAsDown(m)
         updateLatestGossip(newGossip)
         publishMembershipState()
+        if (address == cluster.selfAddress) {
+          // spread the word quickly, without waiting for next gossip tick
+          gossipRandomN(MaxGossipsBeforeShuttingDownMyself)
+        } else {
+          // try to gossip immediately to downed node, as a STONITH signal
+          gossipTo(m.uniqueAddress)
```
Comment (patriknw) [inline akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala:953]: This is what makes biggest difference. Saves around 4 seconds in the tests I tried (downing 12 of 26 nodes).

File: akka-cluster/src/multi-jvm/scala/akka/cluster/StressSpec.scala [hunk 2]
```diff
@@ -59,7 +62,7 @@ private[cluster] object StressMultiJvmSpec extends MultiNodeConfig {
 
   val totalNumberOfNodes =
     System.getProperty("MultiJvm.akka.cluster.Stress.nrOfNodes") match {
-      case null  => 13
+      case null  => 26
```
Comment (patriknw) [inline akka-cluster/src/multi-jvm/scala/akka/cluster/StressSpec.scala:65]: I will revert most changes in this test before merging, but wanted to show what I used for testing this. Downing 12 of 26 nodes.

File: akka-cluster/src/multi-jvm/scala/akka/cluster/StressSpec.scala [hunk 3]
```diff
@@ -687,6 +721,41 @@ abstract class StressSpec
       enterBarrier("remove-several-" + step)
     }
 
+  def partitionSeveral(numberOfNodes: Int): Unit =
```
Comment (patriknw) [inline akka-cluster/src/multi-jvm/scala/akka/cluster/StressSpec.scala:724]: I'll keep this test.

File: akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala [hunk 4]
```diff
@@ -1120,6 +1127,11 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef, joinConfigCompatCh
         coordShutdown.run(CoordinatedShutdown.ClusterLeavingReason)
       }
 
+      if (selfStatus == Down && localGossip.member(selfUniqueAddress).status != Down) {
```
Comment (patriknw) [inline akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala:1130]: Note that MemberDowned event will be published immediately when receiving self as Down. ClusterSingleton and ClusterSharding subscribe to this event and will immediately stop themselves even though the ActorSystem is running some longer.

**Review by johanandren — COMMENTED** [review]
Looks good, esp. the numbers :)

File: akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala (hunk 1 above)
Comment (chbatey) [reply akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala:953]: Nice

File: akka-cluster/src/multi-jvm/scala/akka/cluster/StressSpec.scala (hunk 2 above)
Comment (chbatey) [reply akka-cluster/src/multi-jvm/scala/akka/cluster/StressSpec.scala:65]: 👍

**Review by chbatey — APPROVED** [review]
LGTM!

**Review by johanandren — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
wait, there is some test failure in multi-jvm test

**Comment by patriknw** [comment]
Fixed TransitionSpec. Let me run all tests in jenkins before merging.

**Comment by johanandren** [comment]
@patriknw how is the tests looking for this, can we merge?

**Pull request merged**